### ISSUE TITLE
Downloads are not finishing, sometimes looping infinitely

### ISSUE THREAD

**Issue by EuleMitKeule** [piejanssens/premiumizer#292]
I had this bug on two separate systems now.
Downloads start as expected, but when at 100% they revert back to zero and start again.
The logs tell me there was a "Diff between file sizes".
Sometimes this loop stops after a few tries and then the download stays at 100%, but logs show "I/O Error"
and the .mkv file is still named .mkv.000 as if it were still in the process of downloading.
I am using premiumizer in conjunction with sonarr.

**Comment by EuleMitKeule** [comment]
"02-19 17:06:15: ERROR : [Errno 5] I/O error"
"WARNING : Diff between downloaded files and expected filesizes is 234881024B (filesize: 241231440, expected_filesize: 6350416.0, 1 threads)."

**Comment by neox387** [comment]
This is with pySmartDL
Unfortunately its not the most stable thing. 

You have this always or sometimes ?

**Comment by EuleMitKeule** [comment]
On my old setup I got the infinite loop thing with the "Diff between filesizes" error frequently, but not always.
I just setup a new server with Sonarr and premiumizer and got the I/O error on my first download, after it looped 1 or 2 times. Will test more downloads now and see if I get the I/O error everytime.
I set the UID and GID env variables to the values of the user that owns the sonarr, plex and premiumizer folders on my host.
Permissions for that user are set to rwx so I don't think that should be a problem.
